packages/opentelemetry-core/src/common/attributes.ts

```diff
@@ -47,6 +47,65 @@ export function isAttributeValue(val: unknown): val is AttributeValue {
   return isValidPrimitiveAttributeValue(val);
 }
 
+export function truncateValueIfTooLong(
+  value: AttributeValue,
+  limit: number | undefined,
```

Comment:
are you sure it makes sense for `limit` to actually be optional in a function called `truncateValueIfTooLong`? I've adjusted the type to `number | null`, which seems more suitable

Reply:
yes as then you can call this function with 2 params only and it feels more natural
```
truncateValueIfTooLong('aaa', logger);

// instead of 
truncateValueIfTooLong('aaa', undefined, logger);

// personally I would put logger as 1st param - it is always passed then value, then limit which is connected with value so it feels naturally to be after value.
truncateValueIfTooLong(logger, 'aaa', 123); // truncateValueIfTooLong(logger, 'aaa');
```